### PR TITLE
ACPI local apic info fix

### PR DESCRIPTION
**Pull request by tbursztyka**
First commit fixes an issue seen on UP² where is brings 4 cores (visible in ACPI) but actually has only 2 enabled.

API change got the wrong behavior compared to how it used to work. Fixing this

### CODE REVIEW

**Review by jhedberg — APPROVED** [review]
Tested to work on my Up Squared board. Without this fix most apps fail to boot (basically any app with `CONFIG_SMP=y` which is the default setting).

File: lib/acpi/acpi.c [hunk 1]
```diff
@@ -791,21 +791,29 @@ int acpi_drhd_get(enum AcpiDmarScopeType scope, struct acpi_dmar_device_scope *d
...
 		/* Error on MAD table. */
 		return NULL;
 	}
 
-	if ((cpu_num >= cpu_cnt) || !(lapic[cpu_num].LapicFlags & 1u)) {
-		/* Proccessor not enabled. */
-		return NULL;
+	for (idx = 0; cpu_num >= 0 && idx < cpu_cnt; idx++) {
+		if (lapic[idx].LapicFlags & ACPI_CPU_FLAGS_ENABLED) {
+			if (cpu_num == 0) {
```
Comment (najumon1980) [inline lib/acpi/acpi.c:810]: why you check for BSP only ? This might fail in case of RPL-S where the hard coded x86_cpu_loapics not same.

Comment (tbursztyka) [reply lib/acpi/acpi.c:810]: Why would it fail? We update the x86_cpu_loapics in arch_start_cpu()

**Comment by nashif** [comment]
fix commit message:

```
arch/x86: Remove useless legacy ACPI code

ACPI is not being handled through ACPICA.
```

s/not/now

**Review by najumon1980 — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by finikorg — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Pull request merged**